ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala

```diff
@@ -91,7 +93,10 @@ final case class Claims(
   def validForParticipant(id: String): Boolean =
     participantId.forall(_ == id)
 
-  /** Returns false if the expiration timestamp exists and is greather than or equal to the current time */
+  def validForApplication(id: String): Boolean =
+    applicationId.forall(_ == id)
```

Comment:
Shouldn't this use `exists`? I would expect that `isValidForApplication` is called when the token definitely needs to have a claim for a specific `applicationId`.

Reply:
If the token (the claims) have a missing application ID, then all application IDs are authorized. This is also true for all other properties (ledger ID, participant ID).

Reply:
This would mean that tokens would be forced to include an application identifier and that that token can only be used for that. `None.forall(_ => false)` returns `true`, meaning that in this way we ensure that if a token does not specify a token it can be used for any application identifier (mimicking the same semantics as the `ledgerId` and `participantId` fields checked above).